MDANSE/Src/MDANSE/Framework/Jobs/TrajectoryFilter.py

```diff
@@ -0,0 +1,256 @@
+        """
+        pass
+
+    def finalize(self):
+        """
+        Finalizes the calculations (e.g. averaging the total term, output files creations ...).
+        """
+
+        # Get filter class and instantiate filter object
+        filter_config = json.loads(self.configuration["trajectory_filter"]["value"])
+
+        filter_class, filter_attributes = (
+            filter_map[filter_config["filter"]],
+            filter_config["attributes"],
+        )
+
+        if {"n_steps", "time_step_ps"} not in set(filter_attributes.keys()):
+            filter_attributes.update(
+                {
+                    "n_steps": self.configuration["trajectory"]["length"],
+                    "time_step_ps": self.configuration["trajectory"]["md_time_step"],
+                }
+            )
+
+        filter = filter_class(**filter_attributes)
+
+        trajectories = copy.deepcopy(self.atomic_trajectory_array)
+
+        # Apply filter (only apply initial position offset to atoms if filter is not lowpass and setting has been applied)
+        filtered_coords = apply(
+            filter,
+            trajectories,
+            apply_offsets=filter_attributes.get("attenuation_type", "bandpass")
+            is not "lowpass",
+        )
+
+        # Create trajectory writer object
+        self._output_trajectory = TrajectoryWriter(
+            self.configuration["output_files"]["file"],
+            self.configuration["trajectory"]["instance"].chemical_system,
+            filter_attributes["n_steps"],
+            None,
+            positions_dtype=self.configuration["output_files"]["dtype"],
+            compression=self.configuration["output_files"]["compression"],
+        )
+
+        # Write trajectory
+        write_filtered_trajectory(
+            parent_configuration=self.configuration,
+            nsteps=filter_attributes["n_steps"],
+            filtered_coordinates=filtered_coords,
+            output_trajectory=self._output_trajectory,
+        )
+
+        # The input trajectory is closed.
+        self.configuration["trajectory"]["instance"].close()
+
+        # The output trajectory is closed.
+        self._output_trajectory.close()
+
+        # Write the filter metadata to output
+        outputFile = h5py.File(self.configuration["output_files"]["file"], "r+")
+        outputFile.create_group("metadata").create_dataset(
+            "trajectory_filter",
+            (1,),
+            data=filter.__str__(),
+            dtype=h5py.string_dtype(),
+        )
+
+        outputFile.close()
+
+        super().finalize()
+
+
+def apply(filter, trajectories, apply_offsets: bool) -> np.ndarray:
+    """ """
+    output_trajectory_array = np.zeros(trajectories.shape)
+
+    for at, atom in enumerate(trajectories):
+        x, y, z = atom
+
+        # Store initial positions
+        offsets = np.array([x[0], y[0], z[0]])
+
+        for i, component in ((0, x), (1, y), (2, z)):
```

Comment:
This is just doing `enumerate` manually
```suggestion
        for i, component in enumerate((x, y, z)):
```